### DISCUSSION TITLE
Replicate with many to many relation

### DISCUSSION THREAD

**Discussion by Antonides** [filamentphp/filament#10306] [Help]
Is it possible to replicate the attached options hen you replicate?

`
->beforeReplicaSaved(function (Model $replica): void {
        // Runs after the record has been replicated but before it is saved to the database.
    })
`
Is it possible to get the Model you want to replicate in the beforeReplicaSaved?

**Comment by alkrauss48** [comment]
Yes, it looks like currently (I'm using v3.2.79), you can inject in the original record for all of the 3 replicate lifecycle methods:
```php
Tables\Actions\ReplicateAction::make()
    ->before(function (Model $record): void {
        // Runs before the record has been replicated.
    })
    ->beforeReplicaSaved(function (Model $record, Model $replica): void {
        // Runs after the record has been replicated but before it is saved to the database.
    })
    ->after(function (Model $record, Model $replica): void {
        // Runs after the replica has been saved to the database.
    });
```

From there, you could get the relationships in the `after` function from the `$record`, and attach them to the `$replica`.
